File: Data Structures & Algorithms/is-anagram/submission-1.js
class Solution {
    /**
     * @param {string} s
     * @param {string} t
     * @return {boolean}
     */
    isAnagram(s, t) {
        if (s.length != t.length) {
            return false
        }
        const countS = s.split('').reduce((acc, num) => {
            acc[num] = (acc[num] || 0) + 1
            return acc
        }, {})
        const countT = t.split('').reduce((acc, num) => {
            acc[num] = (acc[num] || 0) + 1
            return acc
        }, {})
                for (const key in countS) {
            if (countS[key] !== countT[key]) {
                return false;
            }
        }
        return true;
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        # [2, 5, 1, 3, 5]
        n = len(nums) - 1
        for num in nums:
            if num < 0:
                num *= -1
            if nums[num - 1] < 0:
                return num
            nums[num - 1] *= -1
        
